src/Stack.spec.ts

```diff
@@ -0,0 +1,70 @@
+import {IRequest, IRequestHeaders} from "./Client";
+import {IMiddleware, Stack} from "./Stack";
+
+describe("Stack", () => {
+    it("should be defined", () => {
+        const stack = new Stack();
+        expect(stack).toBeDefined();
+    });
+    it("Should accept function which returns defaultOptions", () => {
```

Comment:
wrap in describe(construct()) to allow for future expansions (and consistency^^)